go/vt/schemadiff/schema.go

```diff
@@ -750,3 +759,257 @@ func (s *Schema) Apply(diffs []EntityDiff) (*Schema, error) {
 	}
 	return dup, nil
 }
+
+func (s *Schema) ValidateViewReferences() error {
+	var errs error
+	availableColumns := tablesColumnsMap{}
+
+	for _, e := range s.Entities() {
+		entityColumns, err := s.getEntityColumnNames(e.Name(), availableColumns)
+		if err != nil {
+			errs = multierr.Append(errs, err)
```

Comment:
Why not use the Vitess concurrency package here along with `vterrors.Aggregate`? You can see that used throughout the code base if you look for `.AggrError(vterrors.Aggregate)`. Between the `concurrency` and `vterrors` packages we should have whatever related functionality you need here for error handling. For example: https://github.com/vitessio/vitess/blob/390ddad257c7c9ac0ba6cfe15e2a21b0ee943e67/go/vt/wrangler/traffic_switcher.go#L146-L161

Reply:
It looks like it does use locking which is unnecessary here. Dunno if we care about the overhead of that here? That logic seems designed for concurrent error gathering which isn't what we're doing here. 

Using `multierr` seems simpler here and it's already an indirect dependency? 

I think this logic here is temporary anyway, since once https://github.com/golang/go/issues/53435 is available with Go 1.20 we probably want to switch to that anyway.

Reply:
cc @dbussink

Reply:
heh, race condition

Reply:
How strongly do people feel about this? Looking at the two implementation `multierr` does seem to be more fitting to our purpose, but I don't feel strongly.

Reply:
If you only need multiple errors you can use vterrors.Aggregate(), no need for the concurrency piece (as noted, geared toward N goroutines) like we do here e.g.:
https://github.com/vitessio/vitess/blob/3f55e407e8300713d81f7298cdc9677db4be492a/go/vt/vttablet/tabletmanager/vreplication/vcopier.go#L597-L621

I don't feel strongly about it though. Up to you.